rest/src/main/java/com/yahoo/ycsb/webservice/rest/RestClient.java

```diff
@@ -0,0 +1,356 @@
+    conTimeout = Integer.valueOf(props.getProperty(CON_TIMEOUT, "10")) * 1000;
+    readTimeout = Integer.valueOf(props.getProperty(READ_TIMEOUT, "10")) * 1000;
+    execTimeout = Integer.valueOf(props.getProperty(EXEC_TIMEOUT, "10")) * 1000;
+    logEnabled = Boolean.valueOf(props.getProperty(LOG_ENABLED, "false").trim());
+    compressedResponse = Boolean.valueOf(props.getProperty(COMPRESSED_RESPONSE, "false").trim());
+    headers = props.getProperty(HEADERS, "Accept */* Content-Type application/xml user-agent Mozilla/5.0 ").trim()
+          .split(" ");
+    setupClient();
+  }
+
+  private void setupClient() {
+    RequestConfig.Builder requestBuilder = RequestConfig.custom();
+    requestBuilder = requestBuilder.setConnectTimeout(conTimeout);
+    requestBuilder = requestBuilder.setConnectionRequestTimeout(readTimeout);
+    requestBuilder = requestBuilder.setSocketTimeout(readTimeout);
+    HttpClientBuilder clientBuilder = HttpClientBuilder.create().setDefaultRequestConfig(requestBuilder.build());
+    this.client = clientBuilder.setConnectionManagerShared(true).build();
+  }
+
+  @Override
+  public Status read(String table, String endpoint, Set<String> fields, HashMap<String, ByteIterator> result) {
+    int responseCode;
+    try {
+      responseCode = httpGet(urlPrefix + endpoint, result);
+    } catch (Exception e) {
+      responseCode = handleExceptions(e);
+    }
+    if (logEnabled) {
+      System.out.println(new StringBuilder("GET Request: ").append(urlPrefix).append(endpoint)
+            .append(" | Response Code: ").append(responseCode).toString());
+    }
+    return getStatus(responseCode);
+  }
+
+  @Override
+  public Status insert(String table, String endpoint, HashMap<String, ByteIterator> values) {
+    int responseCode;
+    try {
+      responseCode = httpExecute(new HttpPost(urlPrefix + endpoint), values.get("data").toString());
+    } catch (Exception e) {
+      responseCode = handleExceptions(e);
+    }
+    if (logEnabled) {
+      System.out.println(new StringBuilder("POST Request: ").append(urlPrefix).append(endpoint)
+            .append(" | Response Code: ").append(responseCode).toString());
+    }
+    return getStatus(responseCode);
+  }
+
+  @Override
+  public Status delete(String table, String endpoint) {
+    int responseCode;
+    try {
+      responseCode = httpDelete(urlPrefix + endpoint);
+    } catch (Exception e) {
+      responseCode = handleExceptions(e);
+    }
+    if (logEnabled) {
+      System.out.println(new StringBuilder("DELETE Request: ").append(urlPrefix).append(endpoint)
+            .append(" | Response Code: ").append(responseCode).toString());
+    }
+    return getStatus(responseCode);
+  }
+
+  @Override
+  public Status update(String table, String endpoint, HashMap<String, ByteIterator> values) {
+    int responseCode;
+    try {
+      responseCode = httpExecute(new HttpPut(urlPrefix + endpoint), values.get("data").toString());
+    } catch (Exception e) {

```

Comment:
This is catching all exceptions and not actually printing what happened? Would be nice to log the actual exception?